src/mappings/entities/mapToken.ts

```diff
@@ -0,0 +1,165 @@
+import BN from 'bn.js';
+import { SubstrateExtrinsic } from '@subql/types';
+import { ModuleIdEnum, CallIdEnum } from './common';
+import { Token } from '../../types';
+
+const chainNumberMultiplier = new BN(1000000);
+
+export async function mapToken(
```

Comment:
FYI Sonar cloud is saying this function is too big. To make it happy each handler could be its own function and this just routes the event.

Reply:
There's a bunch more stuff left 😁 
Cool, will refactor.